src/gui/guiFormSpecMenu.cpp

```diff
@@ -4376,7 +4376,7 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 
 					ItemStack slct = list_selected->getItem(m_selected_item->i);
 
-					for (s32 i = 0; i < list_s->getSize(); i++) {
+					for (s32 i = 0; i < (s32)list_s->getSize(); i++) {
```

Comment:
What type does `getSize` return? We do not need 32 bits here, should change the type of `i` instead?

Reply:
`getSize` returns an u32.
`InventoryList` (see `inventory.h`) seems to use `u32` everywhere. But there's also for example `MoveAction` (see `inventrymanager.h`), which uses `s16`.
It would probably be good to use `s32` everywhere (after all u16 could get too small for things like creative inventories). But doing that change would require more work.